File: core-android/src/main/java/com/lookaround/core/android/map/clustering/DefaultClusterItem.java
package com.lookaround.core.android.map.clustering;

import androidx.annotation.Nullable;

import com.lookaround.core.android.map.ext.LatLon;

public class DefaultClusterItem implements ClusterItem {
    private final LatLon latLon;

    public DefaultClusterItem(LatLon latLon) {
        this.latLon = latLon;
    }

    @Override
    public double getLatitude() {
        return latLon.getLatitude();
    }

    @Override
    public double getLongitude() {
        return latLon.getLongitude();
    }

    @Nullable
    @Override
    public String getTitle() {
        return null;
    }

    @Nullable
    @Override
    public String getSnippet() {
        return null;
    }
}

File: core-android/src/main/java/com/lookaround/core/android/map/clustering/DefaultClusterItem.kt
package com.lookaround.core.android.map.clustering

import com.lookaround.core.android.map.model.LatLon
import java.util.*

data class DefaultClusterItem(override val uuid: UUID, private val latLon: LatLon) : ClusterItem {
    override val latitude: Double
        get() = latLon.latitude
    override val longitude: Double
        get() = latLon.longitude
    override val title: String?
        get() = null
    override val snippet: String?
        get() = null
}
